### PR TITLE
Add failure scenario handling

### PR DESCRIPTION
**Pull request by jessjenkins**
### What

Added failure scenario handling to import event handler. This updates the image / download variant on the image api when an import failure occurs (cannot read/write from/to s3/vault etc.)

Also minor changes…
- Upgrade go version
- Improve readability of tests (`ShouldHaveLength`)

### How to review

Check that each failure in the import job has an appropriate failure update to the image api. Where this is a failure to the specific download variant (eg. writing a variant to s3) it should update that resource, otherwise it should update the parent image resource (eg. reading the uploaded image key from vault before variants are created).
Ensure there are tests covering the above calls. 
Ensure tests and audit pass.

### Who can review

Anyone but me

### CODE REVIEW

File: event/handler.go [hunk 1]
```diff
@@ -239,3 +248,28 @@ func (h *ImageUploadedHandler) createVaultKeyForFile(keyPath string) ([]byte, er
...
+
+func (h *ImageUploadedHandler) setImageStatusToFailed(ctx context.Context, imageID string, desc string) {
+	image, err := h.ImageCli.GetImage(ctx, "", h.AuthToken, "", imageID)
+	if err != nil {
+		log.Event(ctx, "error getting image from API to set failed_import status", log.ERROR, log.Error(err))
+		return
+	}
+	image.State = failedState
+	image.Error = desc
+	_, err = h.ImageCli.PutImage(ctx, "", h.AuthToken, "", imageID, image)
```
Comment (CarlHembrough) [inline event/handler.go:260]: Consider using a `PATCH` request for this update. This would prevent you having to use a `GET` request before the `PUT`

**Review by CarlHembrough — COMMENTED** [review]
Changes look fine - just one consideration to make about the API call

File: event/handler.go (hunk 1 above)
Comment (nshumoogum) [reply event/handler.go:260]: I think this is a very good point, but requires some more thinking on how this is implemented with the use of IF-Match headers and ETags. I will create a story in the backlog for triage for the tech leads to discuss but this shouldn't hold up this pull request. Thanks Carl

**Review by CarlHembrough — APPROVED** [review]
👍

**Pull request merged**